examples/gno.land/r/demo/boards2/permission_default.gno

```diff
@@ -138,17 +154,16 @@ func (dp *DefaultPermissions) WithPermission(user std.Address, perm Permission,
 func (DefaultPermissions) handleBoardCreate(args Args, cb func(Args)) {
 	// TODO: This way of dealing with arguments is delicate, ideally types should be used
 	name := args[0].(string)
-	if std.Address(name).IsValid() {
-		panic("addresses are not allowed as board name")
-	}
+	assertBoardNameIsNotAddress(name)
+	assertBoardNameBelongsToCaller(name)
 
-	// When the board name is the name of a registered user
-	// check that caller is the owner of the name.
-	caller := std.GetOrigCaller()
-	user := users.GetUserByName(name)
-	if user != nil && user.Address != caller {
-		panic("board name is a user name registered to a different user")
-	}
+	cb(args)
+}
+
+func (DefaultPermissions) handleBoardRename(args Args, cb func(Args)) {
+	newName := args[2].(string)
```

Comment:
Even if this should panic on error, it might be useful to have a descriptive error message.

Can you please add some bounds and type check here?

Reply:
Addressed in 749ab2e